tests/TailwindBuilderTest.php

```
@@ -80,52 +96,59 @@ public function testIntegrationWithMinify(): void
     {
+        $cssFilesSuffix = $this->isTailwindBinaryEqualOrGreaterThan4() ? '-v4' : '';
+
         $builder = new TailwindBuilder(
             __DIR__.'/fixtures',
-            [__DIR__.'/fixtures/assets/styles/app.css', __DIR__.'/fixtures/assets/styles/second.css'],
+            [__DIR__.'/fixtures/assets/styles/app'.$cssFilesSuffix.'.css', __DIR__.'/fixtures/assets/styles/second'.$cssFilesSuffix.'.css'],
             __DIR__.'/fixtures/var/tailwind',
             new ArrayAdapter(),
             null,
-            'v3.4.17',
+            null,
             __DIR__.'/fixtures/tailwind.config.js'
         );
-        $process = $builder->runBuild(watch: false, poll: false, minify: true, inputFile: 'assets/styles/second.css');
+        $process = $builder->runBuild(watch: false, poll: false, minify: true, inputFile: 'assets/styles/second'.$cssFilesSuffix.'.css');
         $process->wait();
 
         $this->assertTrue($process->isSuccessful());
-        $this->assertFileExists(__DIR__.'/fixtures/var/tailwind/second.built.css');
+        $this->assertFileExists(__DIR__.'/fixtures/var/tailwind/second'.$cssFilesSuffix.'.built.css');
 
-        $outputFileContents = file_get_contents(__DIR__.'/fixtures/var/tailwind/second.built.css');
-        $this->assertStringContainsString('body{background-color:blue}', $outputFileContents, 'The output file should contain minified CSS.');
+        $outputFileContents = file_get_contents(__DIR__.'/fixtures/var/tailwind/second'.$cssFilesSuffix.'.built.css');
+        $this->assertStringContainsString('body{background-color:#3b82f6}', $outputFileContents, 'The output file should contain minified CSS.');
     }
 
     public function testIntegrationWithPostcss(): void
     {
+        $cssFilesSuffix = $this->isTailwindBinaryEqualOrGreaterThan4() ? '-v4' : '';
+        if ($cssFilesSuffix) {
+            $this->markTestSkipped('Postcss seems not compatible with Tailwind CLI v4!');
```

Hm, if so I think we should add a note about that in https://symfony.com/bundles/TailwindBundle/current/index.html#using-a-postcss-config-file . WDYT? IIRC Symfony docs have some special syntax for versions, probably we could leverage it to let it know that this does not work since v4 of tailwind

---

I agree. I opened an issue on the TailwindCSS repo and I've got an answer :
```
The CLI does not use postcss anymore so we don't have a plan to making this work. If you rely on postcss I recommend using the [postcss-cli](https://github.com/postcss/postcss-cli) directly with the @tailwindcss/postcss extension.
```
So unfortunately it will not be available in the future.
I'll check the SF docs formats and standards to update it.
Thanks.

---

I just put a warning block in the documentation as the deprecated on seems more appropriate for the bundle version that the binary version. What do you thing about it ?

---

Actually, we could raise and test an exception in case of using PostCSS with v4+ binary. WDYT?

---

hm, throwing an excepting with a clear message explaining that it does not work in Tailwind v4 anymore sounds good to me. Though what happens in the Tailwind binary if when you continue using postcss? Does it also throw an error? Probably the best way would be to do the same to be consistent. But wait, if we just pass the same option to the Tailwind binary, won't it fail upstream? I mean, in theory, we should do nothing because it should fail anyway when the Tailwind binary will run, or am I missing something?

---

No, actually nothing happens! You can add any fake argument to the binary while executing it, no error is thrown and the css is built correctly but without PostCSS processing...